wandb/sdk/launch/runner/kubernetes_runner.py

```diff
@@ -226,6 +243,53 @@ def _watch_job(self) -> None:
                 f"Broken event stream for job watcher in state {state} with selector {self.job_field_selector}: {e}"
             )
 
+    def _watch_crd(self) -> None:
+        """Watch for CRD matching the jobname."""
+        try:
+            for event in self._job_watcher.stream(
+                self.custom_api.list_namespaced_custom_object,
+                namespace=self.namespace,
+                field_selector=self.job_field_selector,
+                group=self.group,
+                version=self.version,
+                plural=self.plural,
+            ):
+                event_type = event.get("type")
+                # This check is needed because CRDs will often not have a status
+                # on the ADDED event, and the status at that time is not useful
+                # in any case.
+                # TODO: Use a constant for these event types.
+                if event_type == "MODIFIED":
+                    object = event.get("object")
+                    status = object.get("status")
+                    state = status.get("state")
+                    if isinstance(state, dict):
+                        state = state.get("phase")
+                    else:
+                        conditions = status.get("conditions")
+                        if isinstance(conditions, list):
+                            if len(conditions) > 0:
+                                state = conditions[-1].get("type")
```

Comment:
Are you certain it will always be in last position?  I couldn't find any reference that guarantees this ordering

Reply:
This was consistently the case with the CRD (kubeflow training jobs) that I added this for, but you are right that this may not be the case for any CRD. I will have to sort conditions based on `lastTransitionTime` first